core/src/main/kotlin/org/quiltmc/qkl/core/annotations/QklDslMarker.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.core.annotations
+
+/**
+ * This marker is for any DSLs that are within the Quilt-Kotlin-Libraries,
+ * to specify that they are DSLs.
+ */
+@DslMarker
+public annotation class QklDslMarker
```

Comment:
Following the brigadier DSL, this should be an annotation unique to this API and not a general purpose one (which might cause issues anyways? not sure)